featurebyte/routes/common/base_table.py

```diff
@@ -174,6 +179,37 @@ async def _add_table_description_from_warehouse(
             )
         return document
 
+    async def _start_table_validation_task(self, table_document: TableDocumentT) -> None:
+        if self.table_facade_service.table_needs_validation(table_document):
+            payload = TableValidationTaskPayload(
+                user_id=self.service.user.id,
+                catalog_id=self.service.catalog_id,
+                table_id=table_document.id,
+                table_name=table_document.name,
+                table_type=table_document.type,
+            )
+            task_id = await self.task_controller.task_manager.submit(payload=payload)
+            table_document = await self.service.get_document(document_id=table_document.id)  # type: ignore[assignment]
+            if (
+                table_document.validation is not None
+                and table_document.validation.status not in TableValidationStatus.terminal()
+            ):
```

Comment:
Curious why don't we overwrite the `validation` attribute directly after we have submit the task?

Reply:
There is a small possibility that the task completed the validation before the update statement here. In unit tests and integration tests, this is always the case because of the mocked task manager.